core/store/src/trie/mem/arena/concurrent.rs

```diff
@@ -0,0 +1,252 @@
+use super::alloc::{allocation_class, allocation_size, CHUNK_SIZE};
+use super::{Arena, ArenaMemory, ArenaPos, ArenaSliceMut, STArena};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+/// Arena that can be allocated on from multiple threads, but still allowing conversion to a
+/// single-threaded `STArena` afterwards.
+///
+/// The `ConcurrentArena` cannot be directly used; rather, for each thread wishing to use it,
+/// `for_thread` must be called to get a `ConcurrentArenaForThread`, which then acts like a
+/// normal arena, except that deallocation is not supported.
+///
+/// The only synchronization they need is the chunk counter. The purpose is so that after
+/// multiple threads allocate on their own arenas, the resulting memory can still be combined
+/// into a single arena while allowing the pointers (ArenaPos) to still be valid. This is what
+/// allows a memtrie to be loaded in parallel.
+pub struct ConcurrentArena {
+    /// Chunks allocated by
```

Comment:
Even with the comment I realised what does concurrent mean only after your answer.

Could you be more specific by saying something like "Index of the next chunk to allocate, shared between `ConcurrentArenaForThread`s, used to collect chunks from all arenas together"?

Reply:
lol looks like I got distracted when writing this comment.